cpp/src/slice2cpp/Gen.cpp

```diff
@@ -2025,6 +2025,12 @@ Slice::Gen::DataDefVisitor::visitExceptionStart(const ExceptionPtr& p)
             H.dec();
             H << sb;
             H << eb;
+
+            // We generate a noexcept copy constructor all the time. A C++ exception must have noexcept copy constructor
```

Comment:
That's the fix for #2358: always generate a noexcept copy constructor even when the fields (e.g. string) don't have one. This will result in a crash if the field copy construction throws.